apps/reader/workers/notification-worker/plugins/publications.ts

```diff
@@ -0,0 +1,54 @@
+import {capture_event} from "posthog";
+import schema from "../../../schema";
+import {EVENT_NAMES, ProcessorPlugin} from "../../replicate-worker/helpers";
+import {NotificationProcessorPlugin, sendNotificationToPostHog} from "../helpers";
+import {NotificationProcessMonitor} from "../monitor";
+
+type PublicationType = 'repost' | 'comment' | 'quote'; 
+
+export class PublicationCreateNotification extends NotificationProcessorPlugin {
+  name(): EVENT_NAMES {
+    return "PublicationCreateWithRef";
+  }
+
+  async process(event: Record<string, any>, monitor: NotificationProcessMonitor, sequence_number: string, signature: string): Promise<void> {
+    // Parse the data we need
+    const parsed = schema.publication_create_with_ref_event_schema.safeParse(event);
+
+    if (!parsed.success) {
+      console.log("Parsing Failed");
+      monitor.setPosthogFailed(sequence_number, parsed.error);
+    } else {
+      const data = parsed.data;
+      console.log(data);
+
+      function getTypeOfPublication(identifier: number): PublicationType | void{
```

Comment:
use the number encoding. 
We don't want to explicitly name the different types.
Also for consistency